### PR TITLE
Add file-content launch substitution

### LINKED ISSUE #707: Add new substitution to read files
As reported here https://github.com/osrf/ros2_test_cases/issues/588#issuecomment-1536261556

### PR DESCRIPTION
**Pull request by cottsay**
This substitution implements a portable mechanism to read the contents of a file on disk. This procedure is used in various scenarios throughout ROS 2, and currently relies on `exec` and `command` substitutions to read files.

I'm particularly interested in review of the substitution name.

Reference implementation: ros2/demos#627

Closes #707

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18712/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13238/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/19445/)

### CODE REVIEW

**Review by mjcarroll — APPROVED** [review]
Great contribution, just one small nit.

File: launch/test/launch/substitutions/test_file_content.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import pytest
+
+
+@pytest.fixture(scope='module')
+def files():
+    this_dir = pathlib.Path(__file__).parent
+
+    files = {
+        'foo': str(this_dir / 'test_file_content' / 'foo.txt'),
+        'bar': str(this_dir / 'test_file_content' / 'bar.txt'),
```
[the remark this answers is not in the document]
Comment (cottsay) [reply launch/test/launch/substitutions/test_file_content.py:32]: `this_dir` is an instance of `pathlib.Path`, so the `/` operator is effectively performing `os.path.join` already.

**Comment by cottsay** [comment]
@mergifyio backport iron

**Comment by emersonknapp** [comment]
@Mergifyio backport humble

**Comment by emersonknapp** [comment]
@Mergifyio backport humble

**Comment by emersonknapp** [comment]
@mergifyio backport humble

**Pull request merged**